pandas/tests/reshape/test_pivot.py

```diff
@@ -1525,116 +1534,3 @@ def test_crosstab_dup_index_names(self):
                                 index=expected_index,
                                 columns=expected_index)
         tm.assert_frame_equal(result, expected)
-
-
-class TestPivotAnnual(object):
-    """
-    New pandas of scikits.timeseries pivot_annual
-    """
-
-    def test_daily(self):
-        rng = date_range('1/1/2000', '12/31/2004', freq='D')
-        ts = Series(np.random.randn(len(rng)), index=rng)
-
-        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
-            annual = pivot_annual(ts, 'D')
-
-        doy = np.asarray(ts.index.dayofyear)
-
-        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
-            doy[(~isleapyear(ts.index.year)) & (doy >= 60)] += 1
-
-        for i in range(1, 367):
-            subset = ts[doy == i]
-            subset.index = [x.year for x in subset.index]
-
-            result = annual[i].dropna()
-            tm.assert_series_equal(result, subset, check_names=False)
-            assert result.name == i
-
-        # check leap days
-        leaps = ts[(ts.index.month == 2) & (ts.index.day == 29)]
-        day = leaps.index.dayofyear[0]
-        leaps.index = leaps.index.year
-        leaps.name = 60
-        tm.assert_series_equal(annual[day].dropna(), leaps)
-
-    def test_hourly(self):
-        rng_hourly = date_range('1/1/1994', periods=(18 * 8760 + 4 * 24),
-                                freq='H')
-        data_hourly = np.random.randint(100, 350, rng_hourly.size)
-        ts_hourly = Series(data_hourly, index=rng_hourly)
-
-        grouped = ts_hourly.groupby(ts_hourly.index.year)
-        hoy = grouped.apply(lambda x: x.reset_index(drop=True))
-        hoy = hoy.index.droplevel(0).values
```

Comment:
anything worthy of a test for these? eg maybe see if something similar using pivot_table (if not can u construct a test)

Reply:
A point is that ``pivot_table`` requires dataframes, and ``pivot_annual`` works on series [that need to have a DatetimeIndex]. So ``pivot_table`` is strictly speaking not doing the same as ``pivot_annual``.

A deeper second issue is that ``pivot_annual`` dayofyear number 60 is **always** febr. 29, while for other pandas date functionality day 60 switches between febr. 29 and 1.mars , depending on whether we're not having a leap year or not.

I'll take a look into if something can be moved. However, do you not think that such date functionality is checked other places, i..e. we duplicate datetime tests by adding them to ``TestPivotTable``? See for example ``TestPivotTable.test_pivot_timegrouper``, line 716 for some date pivot_table tests.